tests/test_catalogue.py

```diff
@@ -7,116 +7,175 @@
 import catalogue.catalogue as ct
 
 
-def test_hash_file(fixtures_dir, empty_hash):
+def test_hash_file(fixtures_dir, copy_fixtures_dir, empty_hash):
 
-    # input is a directory
+    # 1. input is a directory
     with pytest.raises(IsADirectoryError):
         ct.hash_file(fixtures_dir)
 
-    # input is a file
+    # 2. input is a file
     for path, directories, files in os.walk(fixtures_dir):
         for f in files:
             file_path = os.path.join(path, f)
             assert ct.hash_file(file_path).hexdigest() == ct.hash_file(file_path).hexdigest()
             assert ct.hash_file(file_path).hexdigest() != empty_hash
 
-    # input not provided or does not exist
+            # hash of file == hash of file copy
+            copy_file_path = os.path.join(copy_fixtures_dir, f)
+            assert ct.hash_file(file_path).hexdigest() == ct.hash_file(copy_file_path).hexdigest()
+
+    # 3. input not provided or does not exist
     with pytest.raises(TypeError):
         ct.hash_file()
-    with pytest.raises(FileNotFoundError):
+    with pytest.raises(AssertionError):
         ct.hash_file("abc")
+    with pytest.raises(AssertionError):
+        ct.hash_file(123)
 
 
 def test_modified_walk(fixtures_dir, fixture1, fixture2, fixture3, fixture4):
 
+    # valid path
     paths = ct.modified_walk(fixtures_dir)
     assert paths == [fixture1, fixture2, fixture3, fixture4]
 
+    # use ingore_exts
+    paths = ct.modified_walk(fixtures_dir, ignore_exts=[".json"])
+    assert paths == [fixture4]
+
+    paths = ct.modified_walk(fixtures_dir, ignore_exts=[".json", ".csv"])
+    assert paths == []
+
+    paths = ct.modified_walk(fixtures_dir, ignore_exts=[".py"])
+    assert paths == [fixture1, fixture2, fixture3, fixture4]
+
+    # use ignore subdirs
+    base_dir = "tests"
+    subdir = os.path.join(base_dir, "fixtures")
+    paths = ct.modified_walk(base_dir, ignore_subdirs=[subdir])
+    assert (all(
+        [os.path.join(subdir, os.path.basename(fixture)) not in paths
+        for fixture in [fixture1, fixture2, fixture3, fixture4]]))
+
+    # change ignore_dot_files to False
+    paths = ct.modified_walk(".", ignore_dot_files=False)
+    assert "./.gitignore" in paths
 
-def test_hash_dir_by_file(fixtures_dir, fixture1, empty_hash):
+    # path does not exist or not provided
+    with pytest.raises(AssertionError):
+        ct.modified_walk("abc")
+    with pytest.raises(AssertionError):
+        ct.modified_walk(123)
+    with pytest.raises(TypeError):
+        ct.modified_walk()
+
+
+def test_hash_dir_by_file(fixtures_dir, copy_fixtures_dir, fixture1, empty_hash):
```

Comment:
`test_hash_dir_by_file` and `test_hash_dir_full` are largely the same. It would be easier to maintain if you used `@pytest.parametrize` for the shared test cases, and kept non-shared code in the separate functions.

Reply:
I like this idea! I will clean up the tests a bit and then merge.